qt-models/diveplannermodel.cpp

```diff
@@ -1185,26 +1184,10 @@ void DivePlannerPointsModel::saveDuplicatePlan()
 	createPlan(true);
 }
 
-struct divedatapoint *DivePlannerPointsModel::cloneDiveplan(struct diveplan *plan_src, struct diveplan *plan_copy)
+struct divedatapoint *DivePlannerPointsModel::cloneDiveplan(struct diveplan &plan_src, struct diveplan &plan_copy)
```

Comment:
I think now that finding the last segment has become trivial we should be refactoring this to not return the unintuitive `divedatapoint *`, and instead leave fetching of the last segment to the caller.

Reply:
You are right - but it just occurred to me that I made a big mistake:

last_segment should be the third from the back, not the last element. Ooops!

Reply:
Yeah, the naming is slightly less than intuitive. I guess in a plan the second to last segment is the _last segment under water_.
But I stand by my point, let's make this SOLID, and remove the returning of a specific segment from cloning of the plan. Looking at where this is called from, this could probably do with some refactoring as well, and possibly avoid most of the cloning calls.

Reply:
> Looking at where this is called from, this could probably do with some refactoring as well, and possibly avoid most of the cloning calls.

I'm not going to touch that, since I don't understand any of it.